### ISSUE TITLE
Missing Context and Capability Validation in External Service

### ISSUE THREAD

**Issue by volodymyrdovhan** [bambuco/moodle-block_vitrina#17]
Hello,
I found an issue with missing capabilities and context validation in the external services.

Please review the documentation for more details:

[Context and Capability Checks - Moodle External Services](https://moodledev.io/docs/4.5/apis/subsystems/external/writing-a-service#context-and-capability-checks)

**Comment by mudrd8mz** [comment]
Good catch Vlad, thank you for spotting this. I agree this should be fixed and made in line with the documented style.

**Comment by davidherney** [comment]
Hi @volodymyrdovhan ...

I can't use `validate_context` because the method sets the **autologinguest** parameter to false so the block can't display information if the user is not authenticated (including as a guest). Showcase is a block that focuses on displaying information for people who don't have an account yet, and it also displays information on the home page without having logged in as a guest yet.

If  I use `validate_context` in service and I use the block on the site public main page I get this error:
```
requireloginerror
Course or activity not accessible.
```
![error_login](https://github.com/user-attachments/assets/4651bff9-cfbd-4ca0-b19e-cf69929d510b)


So, I want to comply with the requirements but I want to display the courses on the main page even if the person hasn't logged in yet.

**What solution could I adopt then?**
I use `loginrequired = false` but if I use `validate_context` it's not fully considered.

Saludos

**Comment by volodymyrdovhan** [comment]
Hello @davidherney,
the standard flow for such plugins is to enable **Auto-login guests** 

Your plugin perfectly works for Guest users:
![image](https://github.com/user-attachments/assets/3dae13f6-5ff7-4886-a370-928281ee9027)

But maybe you have some other use cases of how the plugin should be used.

@mudrd8mz what do you think about this issue?

Thank
Vlad
